community/kernel/src/main/java/org/neo4j/kernel/api/schema/CompositeIndexDescriptor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.schema;
+
+import org.neo4j.storageengine.api.schema.SchemaRule;
+
+import static java.lang.String.format;
+
+/**
+ * Description of a single index.
```

Comment:
Funny how `single` and `composite` trips over each other when reading this. Also it says nothing about multiple property keys, is it required that a composite index descriptor has multiple keys or can it also handle one? If it can do we need to have multiple classes for this?